yorkie/src/main/kotlin/dev/yorkie/api/ElementConverter.kt

```diff
@@ -70,24 +82,170 @@ internal fun Primitive.toCrdtPrimitive(): CrdtPrimitive {
-        ValueType.VALUE_TYPE_DOUBLE -> PrimitiveType.Double
-        ValueType.VALUE_TYPE_STRING -> PrimitiveType.String
-        ValueType.VALUE_TYPE_BYTES -> PrimitiveType.Bytes
-        ValueType.VALUE_TYPE_DATE -> PrimitiveType.Date
+        PBValueType.VALUE_TYPE_NULL -> PrimitiveType.Null
+        PBValueType.VALUE_TYPE_BOOLEAN -> PrimitiveType.Boolean
+        PBValueType.VALUE_TYPE_INTEGER -> PrimitiveType.Integer
+        PBValueType.VALUE_TYPE_LONG -> PrimitiveType.Long
+        PBValueType.VALUE_TYPE_DOUBLE -> PrimitiveType.Double
+        PBValueType.VALUE_TYPE_STRING -> PrimitiveType.String
+        PBValueType.VALUE_TYPE_BYTES -> PrimitiveType.Bytes
+        PBValueType.VALUE_TYPE_DATE -> PrimitiveType.Date
         else -> error("unimplemented type $this")
     }
 }
 
-internal fun Counter.toCrdtCounter(): CrdtCounter {
-    TODO()
+internal fun PBCounter.toCrdtCounter(): CrdtCounter {
+    TODO("implement after valueFromBytes function")
+}
+
+internal fun PBValueType.toCounterType(): CounterType {
+    return when (this) {
+        PBValueType.VALUE_TYPE_INTEGER_CNT -> CounterType.IntegerCnt
+        PBValueType.VALUE_TYPE_LONG_CNT -> CounterType.LongCnt
+        PBValueType.VALUE_TYPE_DOUBLE_CNT -> CounterType.DoubleCnt
+        else -> error("unimplemented value type: $this")
+    }
 }
 
-internal fun CrdtElement.toJSONElement(): JSONElement {
-    TODO("")
+// TODO(7hong13): should check CrdtText
+internal fun CrdtElement.toPBJsonObject(): PBJsonElement {
+    return when (this) {
+        is CrdtObject -> toPBJsonObject()
+        is CrdtArray -> toPBJsonArray()
+        is CrdtPrimitive -> toPBPrimitive()
+        is CrdtCounter -> toPBCounter()
+        else -> error("unimplemented element: $this")
+    }
+}
+
+internal fun CrdtObject.toPBJsonObject(): PBJsonElement {
+    val pbObject = PBJsonObject.newBuilder().apply {
+        createdAt = this@toPBJsonObject.createdAt.toPBTimeTicket()
+        movedAt = this@toPBJsonObject.movedAt?.toPBTimeTicket()
+        removedAt = this@toPBJsonObject.removedAt?.toPBTimeTicket()
+        rht.toPBRhtNodes().forEachIndexed { index, rhtNode ->
+            setNodes(index, rhtNode)
+        }
+    }.build()
+    return PBJsonElement.newBuilder().apply { jsonObject = pbObject }.build()
+}
+
+internal fun CrdtArray.toPBJsonArray(): PBJsonElement {
+    val pbArray = PBJsonArray.newBuilder().apply {
+        createdAt = this@toPBJsonArray.createdAt.toPBTimeTicket()
+        movedAt = this@toPBJsonArray.movedAt?.toPBTimeTicket()
+        removedAt = this@toPBJsonArray.removedAt?.toPBTimeTicket()
+        elements.toPBRgaNodes().forEachIndexed { index, rgaNode ->
+            setNodes(index, rgaNode)
+        }
+    }.build()
+    return PBJsonElement.newBuilder().apply { jsonArray = pbArray }.build()
+}
+
+internal fun CrdtPrimitive.toPBPrimitive(): PBJsonElement {
+    val pbPrimitive = PBPrimitive.newBuilder().apply {
+        type = this@toPBPrimitive.type.toPBValueType()
```

Comment:
I think all those repeated `this@xx` looks ugly...🤔

Reply:
how about changing like below :)
```
val crdtPrimitive = this
val pbPrimitive = PBPrimitive.newBuilder().apply {
    type = crdtPrimitive.type.toPBValueType()
```